dozent/downloader_tools.py

```diff
@@ -1,100 +1,90 @@
-import multiprocessing
-import os
 import time
-import random
+import sys
+from hurry.filesize import size
 from typing import Tuple
+from threading import Lock
 from pySmartDL import SmartDL
 
+# Used for tracking and displaying download progress bars
+global global_progress_tracker
+global_progress_tracker = [(0, 0, "0", 0, "0")]
+global global_download_size
+global_download_size = 0
+
+global global_final_download_size
+global_final_download_size = 0
+
```

Comment:
Hmm, instead of having these be global, have you considered including them into the `DownloaderTools` class itself?

Reply:
No, because these variables need to be independent of the threads we create. For every thread that's created, an instance of `DownloaderTools` is also created.

Reply:
Hmm, what about class variables in that case?